cdi/src/main/java/io/smallrye/config/inject/ConfigExtension.java

```diff
@@ -184,7 +181,8 @@ protected void validate(@Observes AfterDeploymentValidation adv) {
         try {
             ConfigMappings.registerConfigMappings((SmallRyeConfig) config, configMappingsWithPrefix);
         } catch (ConfigValidationException e) {
-            adv.addDeploymentProblem(e);
+            adv.addDeploymentProblem(InjectionMessages.msg.retrieveConfigPropertiesFailure(e.getLocalizedMessage(), e));
```

Comment:
Is it fair to assume that these validation checks are exclusively for @ConfigProperties property values?

i.e. throwing an exception message saying there was an issue creating a @ConfigProperties bean

Reply:
Yes. This is valid for @ConfigProperties and @ConfigMapping.

Reply:
But probably we should adjust the message, because the message makes a reference to `@ConfigProperties` but the issue might be on a `@ConfigMapping`.

Reply:
Have you got a suggested message that would cover both cases?

Reply:
A possibility would be to group the issues by type in the `ConfigValidationException`. I've also remembered that the `Problem` array may report issues from multiple mappings or properties, meaning that you don't know exactly where the problem occurred (just that it was in the mappings area).

In the Extension we could try to register each mapping / property at a time. That would give us a list of problems per type. And in the `ConfigMappingWithPrefix` we can guess if it is a `@ConfigProperties` or `@ConfigMapping`, by looking into the klass. Or just add that metadata into `ConfigMappingWithPrefix` when it is being constructed.

Reply:
@radcortez do you mean something like:
```
for (ConfigMappings.ConfigMappingWithPrefix cm : configMappingsWithPrefix) {
    try {
        ConfigMappings.registerConfigMappings((SmallRyeConfig) config, new HashSet<>(Arrays.asList(cm)));
    } catch (ConfigValidationException e) {
        if (cm.getKlass().getAnnotation(ConfigMapping.class) != null) {
            adv.addDeploymentProblem(
                    InjectionMessages.msg.retrieveConfigMappingFailure(e.getLocalizedMessage(), e));
        } else if (cm.getKlass().getAnnotation(ConfigProperties.class) != null) {
            adv.addDeploymentProblem(InjectionMessages.msg.retrieveConfigPropertiesFailure(e.getLocalizedMessage(), e));
        } else {
            adv.addDeploymentProblem(e);
        }
    }
}
```
instead of:
https://github.com/smallrye/smallrye-config/blob/c57b5eb12a3360ed137015a99b0adddbcda5eee5/cdi/src/main/java/io/smallrye/config/inject/ConfigExtension.java#L188-L192

Reply:
Yes.

Reply:
This seems like we would be unnecessarily creating new Sets for each element. Perhaps we could add a new `registerConfigMappings(Config, ConfigMappingWithPrefix)` method to `ConfigMappings`?

Reply:
Sure